Improve particle swarm to work in-place (#234)
JuliaManifolds/Manopt.jl: Changelog.md

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).

## [0.4.14] - 06/04/2023

### Changed
* `particle_swarm` now uses much more in-place operations

### Fixed

> * `particle_swarm` used quie a few `deepcopy(p)` commands still, which are now `copy(M,p)`

Reviewer: ```suggestion
* `particle_swarm` used quite a few `deepcopy(p)` commands still, which were replaced by `copy(M, p)`
```
